src/Chainweb/Cut/TestBlockDb.hs

```diff
@@ -0,0 +1,66 @@
+-- |
+-- Module: Chainweb.Cut.TestBlockDb
+-- Copyright: Copyright © 2020 Kadena LLC.
+-- License: MIT
+-- Maintainer: Stuart Popejoy
+--
+-- Maintains block header and payload dbs alongside a current cut.
+--
+
+module Chainweb.Cut.TestBlockDb
+  ( TestBlockDb(..)
+  , withTestBlockDb
+  , addTestBlockDb
+  , getParentTestBlockDb
+  ) where
+
+import Control.Concurrent.MVar
+import Data.Bifunctor (first)
+import qualified Data.HashMap.Strict as HM
+import Data.Tuple.Strict (T2(..))
+
+import Chainweb.BlockHeader
+import Chainweb.ChainId
+import Chainweb.Cut
+import Chainweb.Cut.Test
+import Chainweb.Payload
+import Chainweb.Payload.PayloadStore
+import Chainweb.Payload.PayloadStore.RocksDB
+import Chainweb.Utils
+import Chainweb.Version
+import Chainweb.WebBlockHeaderDB
+
+import Data.CAS
+import Data.CAS.RocksDB
+
+data TestBlockDb = TestBlockDb
+  { _bdbWebBlockHeaderDb :: WebBlockHeaderDb
+  , _bdbPayloadDb :: PayloadDb RocksDbCas
+  , _bdbCut :: MVar Cut
+  }
```

Comment:
That's very interesting. This was the "natural" result of finding the common dependencies of the SPV test, combined with the `MVar Cut` to simply use "current cut" as a shortcut for having a "latest block" concept. 

But now that you mention it, it does resemble `CutDB` (swap a `TVar` in anyway)! 

In general I was very pleased with how easy this refactoring was, which speaks extremely highly of the underlying design.

Reply:
Also I would be happy to do your suggested refactoring. In general I am very glad to be getting to know these parts of the code.

Reply:
Just be aware that it might lead you to the less pleasant territories of the underlying design :-) Namely the code for validating remote cuts and fetching dependencies. But separating that from a local-only cut db might be useful.